R/zchunk_LA1321.cement.R

```diff
@@ -52,72 +63,369 @@ module_energy_LA1321.cement_DISABLED <- function(command, ...) {
+      L1321.IO_Cement_R_limestone_Yh
+
+    # Derive energy inputs to cement production by region and historical year -
+    # Calculate average electric and TPE intensity for each GCAM region (use process emissions as a weighting factor)
+    # ---------------------------------------------------------------------------------------------------------------
+
+    # Interpolate available data on electricity intensity to all historical years
+    IEA_cement_elec_kwht %>%
+      gather(year, value, matches(YEAR_PATTERN)) %>%
+      mutate(year = as.integer(year)) %>%
+      complete(nesting(Country), year = c(year, HISTORICAL_YEARS)) %>%
+      arrange(Country, year) %>%
+      group_by(Country) %>%
+      mutate(value = approx_fun(year, value, rule = 1) * CONV_KWH_GJ / CONV_T_KG) %>%
+      ungroup() ->
+      L1321.IEA_cement_elec_intensity
+
+    # Interpolate available data on total primary energy intensity to all historical years by region
+    IEA_cement_TPE_GJt %>%
+      gather(year, value, matches(YEAR_PATTERN)) %>%
+      mutate(year = as.integer(year)) %>%
+      complete(nesting(Country), year = c(year, HISTORICAL_YEARS)) %>%
+      arrange(Country, year) %>%
+      group_by(Country) %>%
+      mutate(value = approx_fun(year, value, rule = 1) / CONV_T_KG) %>%
+      ungroup() ->
+      L1321.IEA_cement_TPE_intensity
+
+    # Calculate the average electricity generation efficiencies by region to be added to L1321.Cement_ALL_ctry_Yh
+    # -----------------------------------------------------------------------------------------------------------
+
+    # Calculate average regional input energy for electricity across all fuels
+    L123.in_EJ_R_elec_F_Yh %>%
+      group_by(GCAM_region_ID, year) %>%
+      summarise(in.value = sum(value)) %>%
+      ungroup() ->
+      L1321.in_EJ_R_elec_Yh
+
+    # Calculate average regional output energy for electricity across all fuels matching input, join to input energy, and calculate the IO coefficient
+    L123.out_EJ_R_elec_F_Yh %>%
+      semi_join(L123.in_EJ_R_elec_F_Yh, by = "fuel") %>%
+      group_by(GCAM_region_ID, year) %>%
+      summarise(out.value = sum(value)) %>%
+      ungroup() %>%
+      left_join_error_no_match(L1321.in_EJ_R_elec_Yh, by = c("GCAM_region_ID", "year")) %>%
+      mutate(value = in.value / out.value) %>%
+      # NOTE: below replicates the old data system by interpolating IO value to all historical years, but this step can be skipped with current inputs (no new values generated)
+      complete(nesting(GCAM_region_ID), year = c(year, HISTORICAL_YEARS)) %>%
+      arrange(GCAM_region_ID, year) %>%
+      group_by(GCAM_region_ID) %>%
+      mutate(value = approx_fun(year, value)) %>%
+      filter(year %in% HISTORICAL_YEARS) %>%
+      select(GCAM_region_ID, year, value) %>%
+      ungroup() ->
+      L1321.IO_R_elec_Yh
+
+    # Set cap on IO coefficients for regions and years exceeding maximum value - NOTE: Not sure why we have this cap? Worth revisiting.
```

Comment:
Can you open an issue? Also, is it binding in a lot of regions or just 1 or 2?

Reply:
yeah, it affects 116 out of 1280 entries. There's only a few that occur after the year 2000 and most are before 1990. I'll open an issue and leave it alone for now.